frontend/src/plugins/store/auth.js

```diff
@@ -10,13 +12,29 @@ export const mutations = {
   logout(state) {
     state.user = null
   },
-
-  updateUser(state, user) {
-    state.user = user
+}
+export const getters = {
+  /**
+   * @returns The latest Logged-in User Object with a profile()
+   */
+  getLoggedInUserWithProfile: (_authState, getters) => {
+    let loggedInUser = getters.getLoggedInUser
+    if (typeof loggedInUser?.profile !== 'object') return loggedInUser
+    if (!loggedInUser) return loggedInUser
+    let profileHref =
+      loggedInUser._storeData?.profile?.href ?? loggedInUser?.profile?.href
+    let profile = () => apiStore.get(profileHref)
+    return { ...loggedInUser, profile }
+  },
+  getLoggedInUser: (authState, _getters, _rootState, rootGetters) => {
+    if (!authState.user) return authState.user
+    let userFromApiStoreById = rootGetters.getUserFromApiStoreById(authState.user.id)
+    return userFromApiStoreById ? userFromApiStoreById : authState.user
   },
```

Comment:
Our `apiStore` can already automatically do what you are doing in `getLoggedInUser` and `getLoggedInUserWithProfile`, and much more. We noticed a few years ago that the pattern you are implementing here is *very* common, and therefore automated it.

We don't typically work with the data in the Vuex store directly (which you are doing with `getUserFromApiStoreById`). Instead, we use the `apiStore.get` method, to extract the desired data from the Vuex store, or have it automatically be fetched from the API in case it's not already present in the store. The `apiStore.get` method also parses the data and automatically exposes accessor methods for the relations, such as `user.profile()`. When such a relation accessor is used, again it either returns the existing profile data from the store, or dynamically fetches it from the API if needed. So all in all, in Vue components, you can usually just do things like the following, without manually fetching the relation data:
```js
const userProfile = this.user.profile()
const firstActivityTitleOfCamp = this.camp.activities().items[0].title
const displayNamesOfCollaborators = this.camp.campCollaborations().items.map((campCollaboration) => campCollaboration.user().displayName)
// fictional classes for illustration purposes:
const featuredBookEditorialAuthorName = library.featuredBook().editorial().author().name
```

That said, you could delete the `getUserFromApiStoreById` getter, and simplify the code here as follows:
```suggestion
  getLoggedInUser: (authState, _getters, _rootState, rootGetters) => {
    if (!authState.user) return authState.user
    // Get the URI of the user - we use URIs instead of just IDs to identify objects uniquely in the frontend
    userUri = authState.user._meta.self
    
    // Let the API store work its magic. We just give it a URI, and it automagically retrieves the requested
    // data from the store (which acts as a cache), or from the API if it hasn't been fetched before.
    return apiStore.get(userUri)
  },
```

There are some edge cases to do with loading entities. If you run into issues while some data is still being fetched from the API but already displayed in the UI, let me know so I can explain the options the apiStore has for these cases. (In short: Every entity and every related entity have the `._meta.loading` boolean flag and the `._meta.load` Promise which can be `await`ed)

Reply:
All this automated logic (parsing the API responses, exposing relation accessors, loading attributes, and even advanced features such as templated links and methods to POST, PATCH and DELETE to the API) are extracted and documented in this repository (which we published as an NPM package): https://github.com/ecamp/hal-json-vuex/tree/next